### PR TITLE
Protobuf

### PR DESCRIPTION
**Pull request by tallytalwar**
Parsing speedup: 15-20% compared to rapid json
Network access speedup: 50%

### CODE REVIEW

**Comment by blair1618** [comment]
There's some spurious geometry being generated here, haven't determined the cause yet.

Protobuf:
protobuff

Master (rapidjson):
master

**Comment by tallytalwar** [comment]
Interesting. Looks like more polys...

File: core/src/util/pbfParser.cpp [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    pbfGeomCmd cmd = pbfGeomCmd::moveTo;
+    uint32_t cmdRepeat = 0;
+    protobuf::message geomItr = _in;
+    
+    double invTileExtent = (1.0/(double)_tileExtent);
+    
+    std::vector<Point> line;
+    
+    int x = 0.0f;
+    int y = 0.0f;
```
Comment (blair1618) [inline core/src/util/pbfParser.cpp]: ?

File: core/src/util/pbfParser.cpp [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+                            break;
+                        case 7: // bool value
+                            numericValues[valueCount] = valueItr.boolean();
+                            break;
+                        default:
+                            valueItr.skip();
+                            break;
+                    }
+                }
+                valueCount++;
```
Comment (blair1618) [inline core/src/util/pbfParser.cpp]: Seems like this should be in the scope immediately above this

**Pull request merged**